apps/dapp-console/app/components/Auth/RequireLogin.tsx

```diff
@@ -0,0 +1,52 @@
+import { useLogin, useLogout, usePrivy } from '@privy-io/react-auth'
+import { apiClient } from '@/app/helpers/apiClient'
+import { toast } from '@eth-optimism/ui-components'
+import { LONG_DURATION } from '@/app/constants/toast'
+import { captureError } from '@/app/helpers/errorReporting'
+import { redirect } from 'next/navigation'
+import { useEffect, useState } from 'react'
+
+export const RequireLogin = ({ children }: { children: React.ReactNode }) => {
+  const { authenticated } = usePrivy()
+  const { mutateAsync: loginUser } = apiClient.auth.loginUser.useMutation()
+  const { mutateAsync: logoutUser } = apiClient.auth.logoutUser.useMutation()
+  const [pendingLogin, setPendingLogin] = useState(false)
+  const { login } = useLogin({
+    onComplete: async () => {
+      try {
+        await loginUser()
+      } catch (e) {
+        toast({
+          description: 'Sign in failed.',
+          duration: LONG_DURATION,
+        })
+        captureError(e, 'loginUser')
+        logout()
+      }
+    },
+    onError: (privyError) => {
```

Comment:
this callback gets called when there are errors in the privy auth flow or when the user exits out of the login modal.